opentracing/span.go

```diff
@@ -0,0 +1,165 @@
+package opentracing
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	opentracing "github.com/opentracing/opentracing-go"
+	"sourcegraph.com/sourcegraph/appdash"
+)
+
+// Span is the Appdash implementation of the `opentracing.Span` interface.
+type Span struct {
+	sync.Mutex
+	Recorder      *appdash.Recorder
+	tracer        *Tracer
+	operationName string
+	startTime     time.Time
+	sampled       bool
+	attributes    map[string]string
+	tags          map[string]interface{}
+
+	logs []opentracing.LogData
+}
+
+func newAppdashSpan(operationName string, tracer *Tracer) *Span {
+	return &Span{
+		operationName: operationName,
+		tracer:        tracer,
+		startTime:     time.Now(),
+		logs:          make([]opentracing.LogData, 0),
+	}
+}
+
+// SetOperationName sets the name of the span, overwriting the previous value.
+func (s *Span) SetOperationName(operationName string) opentracing.Span {
+	s.operationName = operationName
+	return s
+}
+
+// Tracer returns the interal opentracing.Tracer
+func (s *Span) Tracer() opentracing.Tracer {
+	return s.tracer
+}
+
+// Finish ends the span.
+//
+// Defers to s.FinishWithOptions() with FinishTime = time.Now()
```

Comment:
style nit (applies to opentracing-go pkg too):

Using the wording `Defers` is strange because Go has a `defer` keyword. Typically, this would be written more concisely as e.g.:

```
// Finish ends the span. It is short-handed for:
//
//  s.FinishWithOptions(opentracing.FinishOptions{FinishTime: time.Now()})
//
```